chapters/classes.tex

```diff
@@ -276,36 +276,91 @@ \subsubsection{Prefix Rules}\label{prefix-rules}
+The unexpanded binding equations for constants in the translated model must be acyclic after flattening.
+Thus it is not possible to introduce equations for constants by cyclic dependencies.
 
-\begin{nonnormative}
-There is no exception for parameters with \lstinline!fixed = false!, despite the fact that such parameters are generally allowed to be initialized from systems of dependent equations.
-% Saying "binding equation" below even though it's called a "declaration equation", for consistency with the normative paragraph above.
-However, a parameter with \lstinline!fixed = false! can use an initial equation instead of a binding equation, allowing for cyclic dependencies.
-\end{nonnormative}
+Structural parameters, on the other hand, cannot have cyclic dependencies by definition.
+Since variability rules do not allow constants to depend on structural parameters, this means that constants and structural parameters share the property that their declaration equations can be solved one at a time with the natural causality.
 
 \begin{example}
 \begin{lstlisting}[language=modelica]
 constant Real p = 2 * q;
 constant Real q = sin(p); // Illegal since p = 2 * q, q = sin(p) are cyclical
+\end{lstlisting}
+\end{example}
+
+\subsection{Component Variability Prefixes discrete, parameter, constant}\label{component-variability-prefixes-discrete-parameter-constant}
+
+The prefixes \lstinline!discrete!, \lstinline!parameter!, \lstinline!constant! of a component declaration
+are called variability prefixes and define in which situation the
+variable values of a component are initialized (see \cref{events-and-synchronization} and
+\cref{initialization-initial-equation-and-initial-algorithm}) and when they are changed in transient analysis (= solution
+of initial value problem of the hybrid DAE):
+\begin{itemize}
+\item
+  A variable \lstinline!vc! declared with \lstinline!constant! prefix remains constant during transient analysis, with a value that is unaffected by the initialization problem (i.e., determined during translation).
+\item
+  % TODO: Add "structural parameter" to index.
+  A variable \lstinline!vc! is called a \firstuse{structural parameter} if all of the following applies:
+  \begin{itemize}
+  \item
+    It is declared with the \lstinline!parameter! prefix.
+  \item
+    It has \lstinline!fixed = true!.
+  \item
+    It does not have annotation \lstinline!Evaluate = false!.
+  \item
+    The expression of its declaration equation does not depend on \lstinline!vc! itself, neither directly nor via cyclic dependency on other parameters.
+    To satisfy this condition, dependencies shall be removed as needed by applying simplifications based on values of constants and other structural parameters that don't depend on \lstinline!vc! with \lstinline!Evaluate = true!.
+    % What about breaking cycles by simplification based on values of other evaluated structural parameters that don't have Evaluate = true?
```

Comment:
The definition would be better – easier to reason about and to implement consistently across tools – done this way:
```suggestion
    To satisfy this condition, dependencies shall be removed as needed by applying simplifications based on values of constants (except with \lstinline!Evaluate = false!)  and all other structural parameters that don't depend on \lstinline!vc!.
    That the value of a structural parameter is used for these simplifications does not mean that it has to be determined during translation, but if \lstinline!vc! is found to be a structural parameter, then a Modelica tool will be able to break all cycles involving \lstinline!vc! by making some (possibly none or all) of the other structural parameters determined during translation.
```

Reply:
Since several concerns have been raised already, but none of them regarding this change, I'm applying it now.  Then, we can discuss it further as part of the (new) current state of the proposal.